app/services/invoices/create_service.rb

```diff
@@ -2,36 +2,41 @@
 
 module Invoices
   class CreateService < BaseService
-    def initialize(subscription:, timestamp:)
-      @subscription = subscription
+    def initialize(subscriptions:, timestamp:)
+      @subscriptions = subscriptions
       @timestamp = timestamp
+      @customer = subscriptions&.first&.customer
+      @currency = subscriptions&.first&.plan&.amount_currency
 
       super(nil)
     end
 
     def create
       ActiveRecord::Base.transaction do
-        invoice = Invoice.find_or_create_by!(
-          subscription: subscription,
-          from_date: from_date,
-          to_date: to_date,
-          charges_from_date: charges_from_date,
-          issuing_date: issuing_date,
+        invoice = Invoice.create!(
+          customer: customer,
+          issuing_date: (Time.zone.at(timestamp) - 1.day).to_date,
```

Comment:
I talked with Raffi and we should probably simplify logic for issuing date and use invoice creation date here